### DISCUSSION TITLE
getting sender address in rcpt hook plugin

### DISCUSSION THREAD

**Discussion by rickersilva** [haraka/Haraka#3157] [Q&A] [answered]
I'm trying to write a custom plugin in which I need to obtain both sender address and recipient address in haraka incoming mail.

I'm using the swaks command to send a mail to my haraka server, so the -f attribute is a gmail account and the -t account is the haraka mail.

And I'm using the usual code in the sample plugin to retrieve the recipient: 
` var rcpt = params[0];`

I was browsing connection attributes and found this for the sender email, but I want to be sure this is a good practice and the right way to do it.
`var sender = connection.transaction.mail_from;`

Is this correct way to retrieve the sender email? could I face troubles down the road in production environments?

**Comment by msimerson** [comment, marked as answer]
that looks right to me.
